workers/unity/Assets/Playground/Scripts/MonoBehaviours/SpawnCubeCommandReceiver.cs

```diff
@@ -0,0 +1,112 @@
+﻿using System.Collections.Generic;
+using Generated.Improbable;
+using Generated.Playground;
+using Improbable.Gdk.Core;
+using Improbable.Gdk.Core.Commands;
+using Improbable.Gdk.Core.GameObjectRepresentation;
+using Improbable.Worker.Core;
+using UnityEngine;
+using Transform = Generated.Improbable.Transform.Transform;
+
+#region Diagnostic control
+
+// Disable the "variable is never assigned" for injected fields.
+#pragma warning disable 649
+
+// ReSharper disable PossibleInvalidOperationException
+
+#endregion
+
+namespace Playground.MonoBehaviours
+{
+    public class SpawnCubeCommandReceiver : MonoBehaviour
+    {
+        [Require] private Transform.Requirables.Reader transformReader;
+        [Require] private CubeSpawner.Requirables.CommandRequestHandler commandRequestHandler;
+        [Require] private CubeSpawner.Requirables.Writer writer;
+        [Require] private WorldCommands.Requirables.WorldCommandRequestSender worldCommandRequestSender;
+        [Require] private WorldCommands.Requirables.WorldCommandResponseHandler worldCommandResponseHandler;
+
+        private ILogDispatcher logDispatcher;
+
+        private const string EntityForIdFailedWithMessage = "Create entity (for id {0}) failed with message: \"{1}\"";
+        private const string FailedToReserveEntityIdMessage = "Failed to reserve entity id: {0}";
+
+        private readonly HashSet<long> sentRequestIds = new HashSet<long>();
+
+        public void OnEnable()
+        {
+            commandRequestHandler.OnSpawnCubeRequest += OnSpawnCubeRequest;
+
+            worldCommandResponseHandler.OnReserveEntityIdsResponse += OnEntityIdsReserved;
+            worldCommandResponseHandler.OnCreateEntityResponse += OnEntityCreated;
+
+            logDispatcher = GetComponent<SpatialOSComponent>().LogDispatcher;
+        }
+
+        private void OnSpawnCubeRequest(CubeSpawner.SpawnCube.RequestResponder requestResponder)
+        {
+            requestResponder.SendResponse(new Empty());
+
+            sentRequestIds.Add(worldCommandRequestSender.ReserveEntityIds(1));
+        }
+
+        private void OnEntityIdsReserved(WorldCommands.ReserveEntityIds.ReceivedResponse response)
+        {
+            if (!sentRequestIds.Remove(response.RequestId))
+            {
+                // This response was not for a command from this behaviour.
```

Comment:
Are we not filtering this at a layer above this? Or do we have multiple behavours per entity? Or is this trying to solve a race where the entity doing this is checked in then back out while a request was in progress (which I think core already handles gracefully)?

Reply:
It can potentially be picked up by this if this entity is used as a command sender by another behaviour, or even through ECS.

The filtering checks that the entity has the responders injected but not "what thing sent the request" - as potentially it could be something the users may want to handle like split sending / responding to different behaviours or even gradually move to/from ECS; I'm stretching my imagination quite far here though